From e3f29547b0dcef329a74eeca0be18ae18f1b780b Mon Sep 17 00:00:00 2001
From: Pierre K <pierrekn@gmail.com>
Date: Sun, 6 Nov 2022 20:55:25 +0100
Subject: [PATCH 176/196] [test] fix Python 3.10 asyncio DeprecationWarning
 (#278)

With Python 3.10, calling `asyncio.get_event_loop().run_until_complete`  raises a `DeprecationWarning`, which is interpreted as a failure by the test framework because there should be nothing on stderr.

With Python 3.11, this will become a `RuntimeError`.

Just create the loop before running it.
---
 test/functional/interface_nng.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/functional/interface_nng.py b/test/functional/interface_nng.py
index 7895e0a27..497ced449 100755
--- a/test/functional/interface_nng.py
+++ b/test/functional/interface_nng.py
@@ -57,7 +57,8 @@ class NngInterfaceTest(BitcoinTestFramework):
         self.anyone_script = self.nodes[0].validateaddress(self.anyone_addr)['scriptPubKey']
         self.anyone_addr2 = self.nodes[0].decodescript('52')['p2sh']
         self.anyone_script2 = self.nodes[0].validateaddress(self.anyone_addr2)['scriptPubKey']
-        asyncio.get_event_loop().run_until_complete(self._nng_test())
+        loop = asyncio.new_event_loop()
+        loop.run_until_complete(self._nng_test())
 
     async def _nng_test(self):
         import pynng, flatbuffers
-- 
2.39.5 (Apple Git-154)

